README.md

```diff
@@ -9,3 +9,34 @@ The `materialize-*` connectors all implement the [Flow Materialize gRPC protocol
+- [ ] For SQL captures, you can use the [sqlcapture](https://github.com/estuary/connectors/tree/main/sqlcapture) library. For an example, see [source-postgres](https://github.com/estuary/connectors/tree/main/source-postgres).
+- [ ] When emitting date-time values, if the discovered schema of the connector emits `format: date-time`, the value must be RFC3339 compliant, or otherwise parsing of the value will fail.
+
+## Materialization Connectors
+
+- [ ] Check out the protocol definitions of Flow, they include a lot of comments explaining the various interfaces and messages used to communicate between your connector and Flow runtime: [materialize.proto](https://github.com/estuary/flow/blob/master/go/protocols/materialize/materialize.proto)
+- [ ] You need to choose the right pattern for materialization based on transactional guarantees of your destination technology. This choice is important to uphold exactly-once semantics expected from most connectors. See the comments in [materialize.proto](https://github.com/estuary/flow/blob/master/go/protocols/materialize/materialize.proto) for more technical details.
+  - [ ] If your technology supports committing all data during store phase as part of a transaction, then you can have your destination be authoritative about the checkpoint by also updating the checkpoint as part of the store phase transaction. For an example of this see [materialize-postgres](https://github.com/estuary/connectors/tree/main/materialize-postgres).
+  - [ ] If your technology does not support transactions, but does support a retriable idempotent store operation, then you can have the Flow Recovery Log be authoritative and use the idempotency of the operation to ensure exactly-once semantics even in cases of failure. The general idea is to keep track of the operations that are being run in the store phase in the checkpoint sent to the runtime, so that in case the connector fails before it can successfully commit the checkpoint to the recovery log, on the next start of the connector, the operation can be retried by looking at the checkpoint and running the idempotent operations again as part of the Open phase. For an example of this see [materialize-databricks](https://github.com/estuary/connectors/tree/main/materialize-databricks).
+  - [ ] There are certain technologies that will not support either of these, in those cases we cannot guarantee exactly-once semantics, but at-least-once. For an example of this see [materialize-google-pubsub](https://github.com/estuary/connectors/tree/main/materialize-google-pubsub).
+- [ ] Note that in a mater
+- [ ] If your technology supports different modes of authentication, the recommended approach is to have a `oneOf` jsonschema under the key `credentials`. See [materialize-databricks](https://github.com/estuary/connectors/blob/main/materialize-databricks/.snapshots/TestSpecification) as an example.
```

Comment:
This is a really good idea and I wish we had done it more with existing connectors. I'd almost say that even if the technology doesn't (yet) support different modes of operation, to still strongly consider a `oneOf`, just in case a new mode is added or discovered